### PR TITLE
Issue 10255 improve coach tabs accessibility

### PR DESCRIPTION
**Pull request by muditchoudhary**
## Summary

Use KDS tabs (KTabsList and KTabsPanel) in the Learners tab which is inside Coach->Reports->Learners->Select a learner.
Add new constants for the Learners tab inside `tabsConstants.js`

## References

#10255

## Reviewer guidance

To review

Go to Coach->Reports-Learners-Select a learner
For accessibility, use the keyboard and go to the **Reports tab** and then press the **right arrow** to move focus to the **Activity tab**. Press enter to open the focused tab.
For normal, use the mouse to click the respective tab to open.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by muditchoudhary** [comment]
Hi @MisRob Could you please review the code?

**Comment by MisRob** [comment]
Hi @muditchoudhary, thank you! I will have a look some time this week

**Comment by muditchoudhary** [comment]
> will have a look some time this week

Sure!

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerHeader.vue [hunk 1]
```diff
@@ -53,13 +53,11 @@
...
-        :text="coachString('reportsLabel')"
-        :to="classRoute('ReportsLearnerReportPage', {})"
-      />
-      <HeaderTab
-        :text="coachString('activityLabel')"
-        :to="classRoute('ReportsLearnerActivityPage', {})"
+      <KTabsList
+        :tabsId="LEARNERS_TABS_ID"
+        :activeTabId="activeTabId"
+        ariaLabel="Coach learners"
```
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerHeader.vue]: "Coach learners" will need to be translated, similarly to how `"coachString('reportsLabel')"` is. Let me know if you needed some guidance in regards to translations.

**Review by MisRob — COMMENTED** [review]
Hi @muditchoudhary, overall this is great work. Code changes make sense to me and I can confirm that the basic functionality works as expected.

At this point, I only see one thing to be tuned. Here, you can see that after I press enter on the Activity tab, the focus ring gets lost after the tab content gets loaded:

lost-focus

However, it should stay visible, in the same way it stays visible when I press enter on Quizzes tab here, for example:

focus

I assume that the problem is that the whole page gets reloaded after pressing enter and therefore the focus is lost. This is something I needed to deal in https://github.com/learningequality/kolibri/pull/10212 too. I prepared [useCoachTabs.js](https://github.com/learningequality/kolibri/pull/10212/files#diff-f5d2116969b4368b118690b9560a2484073765c6e435a15aaeebb1fbbb6c7e4e) utilities. You could read its documentation and then see in the codebase how it's used. This is a bit tricky part. Let me know if you needed anything.

**Comment by MisRob** [comment]
@radinamatic I will invite you to test as soon as we arrive at a final version

**Comment by muditchoudhary** [comment]
Hi @MisRob I fixed the ring issue. Could you please check the PR? Thanks!!
Peek 2023-11-22 20-32

**Comment by MisRob** [comment]
Thanks @muditchoudhary, it's looking great! I will re-review some time next week.

**Comment by MisRob** [comment]
@muditchoudhary Meanwhile, if you could fix linting, that'd be good

**Comment by muditchoudhary** [comment]
> @muditchoudhary Meanwhile, if you could fix linting, that'd be good

Sure! I forgot to run the lint command before the push

**Comment by muditchoudhary** [comment]
> Thanks @muditchoudhary, it's looking great! I will re-review some time next week.

okay

File: kolibri/plugins/coach/assets/src/constants/tabsConstants.js [hunk 2]
```diff
@@ -19,3 +19,9 @@ export const ReportsGroupTabs = {
   MEMBERS: 'tabMembers',
   ACTIVITY: 'tabActivity',
 };
+
+export const LEARNERS_TABS_ID = 'reportLearners';
```
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/constants/tabsConstants.js:23]: Just a nitpick, no change required. May be nice to follow the naming convention of other constants and use `'coachReportsLearners'` as ID value.

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerHeader.vue [hunk 3]
```diff
@@ -115,13 +133,43 @@
...
+          },
+        ];
+      },
+    },
+    mounted() {
+      // focus the active tab but only when it's likely
+      // that this header was re-mounted as a result
+      // of navigation after clicking a tab (focus shouldn't
+      // be manipulated programatically in other cases, e.g.
+      // when visiting the Plan page for the first time)
```
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerHeader.vue:156]: Minor comment: "when visiting the Plan page" This is not the Plan page. I think that's result of copy-pasting, so perhaps we could just say "when visiting the page" to make future copies work without the need to adjust it

File: kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerHeader.vue [hunk 4]
```diff
@@ -115,13 +133,43 @@
...
     },
     $trs: {
       back: {
         message: 'All learners',
         context:
           "Link that takes user back to the list of learners on the 'Reports' tab, from the individual learner's information page.",
       },
+      reportLearners: {
+        message: 'Report learners',
+        context: 'Labels the Reports > Learners tab for screen reander users',
```
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/views/reports/ReportsLearnerHeader.vue:171]: A little typo: "...screen reader"

**Review by MisRob — APPROVED** [review]
Amazing work @muditchoudhary. This was lots of work and definitely not a trivial task due to how we do routing and you resolved it nearly on your own.

Code is okay. I left very minor comments, by no means anything blocking.

I also tested manually and can confirm that it works as expected. @radinamatic would you like to QA it or is my testing sufficient here as this targets `develop`?

**Comment by muditchoudhary** [comment]
> Amazing work @muditchoudhary. This was lots of work and definitely not a trivial task due to how we do routing and you resolved it nearly on your own.

Thank you!! The code and files are organized well and easy to read + great documentation. That is why I can understand it.

**Review by pcenov — APPROVED** [review]
Thank you @muditchoudhary - I confirm that this is implemented as specified.

**Comment by muditchoudhary** [comment]
Hello, @MisRob I forgot to tell to wait before merging the PR so I can resolve the minor fix too that you mentioned. Now since the branch is merged, I'll raise a new PR, hope that will be fine.

**Comment by MisRob** [comment]
Ah I'm sorry about that @muditchoudhary. We are excited to have it merged, I guess :)! If you don't mind opening a new PR, please go ahead! Or you could also do it in your other work in progress - just adding a few commits, if that'd be easier.

**Comment by muditchoudhary** [comment]
> Ah I'm sorry about that @muditchoudhary. We are excited to have it merged, I guess :)! If you don't mind opening a new PR, please go ahead! Or you could also do it in your other work in progress - just adding a few commits, if that'd be easier. 

No issues for me. On my next PR I'll also include these. That would be easier for me. 
Thanks

**Pull request merged**